src/ansible_compat/runtime.py

```diff
@@ -68,22 +96,301 @@ def version(self) -> Version:
             msg = "Unable to find a working copy of ansible executable."
             raise MissingAnsibleError(msg, proc=proc)
 
-        self._version = Version(version)
+        self._version = packaging.version.Version(version)
         return self._version
 
-    # def install_collection(self, collection: str) -> None:
-    #     """..."""
-    #     ...
-
-    # def install_requirements(self, requirement: str) -> None:
-    #     """..."""
-    #     ...
-
-    # def prepare_environment(
-    #     self,
-    #     project_dir: Optional[str] = None,
-    #     offline: bool = False,
-    #     required_collections: Optional[Dict[str, str]] = None,
-    # ) -> None:
-    #     """..."""
-    #     ...
+    def install_collection(
+        self, collection: str, destination: Optional[Union[str, pathlib.Path]] = None
+    ) -> None:
+        """Install an Ansible collection.
+
+        Can accept version constraints like 'foo.bar:>=1.2.3'
+        """
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
+            "-v",
+        ]
+        if destination:
+            cmd.extend(["-p", str(destination)])
+        cmd.append(f"{collection}")
+
+        _logger.info("Running %s", " ".join(cmd))
+        run = subprocess.run(
+            cmd,
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+        if run.returncode != 0:
+            _logger.error("Command returned %s code:\n%s", run.returncode, run.stdout)
+            raise InvalidPrerequisiteError()
+
+    @tenacity.retry(  # Retry up to 3 times as galaxy server can return errors
+        reraise=True,
+        wait=tenacity.wait_fixed(30),  # type: ignore
+        stop=tenacity.stop_after_attempt(3),  # type: ignore
+        before_sleep=tenacity.after_log(_logger, logging.WARNING),  # type: ignore
+    )
+    def install_requirements(self, requirement: str) -> None:
+        """Install dependencies from a requirements.yml."""
+        if not os.path.exists(requirement):
+            return
+
+        cmd = [
+            "ansible-galaxy",
+            "role",
+            "install",
+            "--roles-path",
+            f"{self.cache_dir}/roles",
```

Comment:
It's usually best to couple options and their values on the same line because otherwise, they look standalone which adds more cognitive complexity for the reader to figure out which ones are related and separate those that aren't.